frontend/public/components/app.jsx

```diff
@@ -238,6 +240,40 @@ const AppRouter = () => {
   );
 };
 
+const CaptureTelemetry = React.memo(() => {
+  const fireTelemetryEvent = useTelemetry();
+
+  // notify of identity change
+  const user = useSelector(({ UI }) => UI.get('user'));
+  React.useEffect(() => {
+    if (user.metadata?.uid || user.metadata?.name) {
+      fireTelemetryEvent('identify', { user });
+    }
+    // Only trigger identify event when the user identifier changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [user.metadata?.uid || user.metadata?.name, fireTelemetryEvent]);
```

Comment:
```suggestion
  }, [user?.metadata.uid || user?.metadata.name, fireTelemetryEvent]);
```

Reply:
If we make this change here we will get `TypeError: Cannot read property 'uid' of undefined`.

Reply:
> If we make this change here we will get `TypeError: Cannot read property 'uid' of undefined`.

That worries me to honest. `metadata` should always be defined on a user. Is user getting initialized to `{}` maybe?

Reply:
I am not sure. @christianvogt Any thought of this?